allennlp/models/semantic_parsing/quarel/quarel_semantic_parser.py

```diff
@@ -15,7 +15,7 @@
 from allennlp.semparse.type_declarations import type_declaration
 from allennlp.semparse.type_declarations.type_declaration import START_SYMBOL
 from allennlp.semparse.worlds.quarel_world import QuarelWorld
-from allennlp.semparse import ParsingError
+from allennlp.semparse.common.errors import ParsingError
```

Comment:
You've imported `ParsingError` in `allennlp.semparse.__init__`, so you don't actually need to change this line (same for several other files).

Reply:
Something weird was happening before changing all the imports to more specific imports such as this.
Same kind of error, but I don't know why.
```

In [1]: from allennlp.semparse.contexts.table_question_context import TableQuestionContext                                                                                                                                 
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-298ceba81f69> in <module>
----> 1 from allennlp.semparse.contexts.table_question_context import TableQuestionContext

...

~/allennlp/allennlp/semparse/contexts/table_question_context.py in <module>
      6 from unidecode import unidecode
      7 from allennlp.data.tokenizers import Token
----> 8 from allennlp.semparse.domain_languages.common import Date
      9 from allennlp.semparse.contexts.knowledge_graph import KnowledgeGraph
     10 

~/allennlp/allennlp/semparse/domain_languages/__init__.py in <module>
----> 1 from allennlp.semparse.domain_languages.domain_language import (DomainLanguage, START_SYMBOL,
      2                                                                 predicate, predicate_with_side_args)
      3 from allennlp.semparse import ParsingError, ExecutionError
      4 from allennlp.semparse.domain_languages.nlvr_language import NlvrLanguage
      5 from allennlp.semparse.domain_languages.quarel_language import QuaRelLanguage

~/allennlp/allennlp/semparse/domain_languages/domain_language.py in <module>
     10 
     11 from allennlp.common.util import START_SYMBOL
---> 12 from allennlp.semparse import util, ParsingError, ExecutionError
     13 
     14 logger = logging.getLogger(__name__)

ImportError: cannot import name 'ParsingError' from 'allennlp.semparse' (/data/nilesh/allennlp/allennlp/semparse/__init__.py)
```

Reply:
If the file is underneath `allennlp/semparse/`, you need the fully-qualified path, or you get circular import problems.  If the file is outside of it, it should be ok to just do `from allennlp.semparse import ParsingError`.

Reply:
I should've broken this into two commits.

I've committed changes sans importing from more specific package (so, without changes like this, but only moving Date, errors, importing from semparse) here: https://github.com/nilesh-c/allennlp/tree/dateimportfix2

Reply:
AH. Alright, I'll fix these up and make a push tomorrow.

Reply:
Reverted back the imports outside `allennlp.semparse` except for `allennlp/data/dataset_readers/semantic_parsing/wikitables/wikitables.py` because `data` has circular dependencies.